R/MsBackendTimsTof-functions.R

```diff
@@ -88,4 +88,35 @@
 
 MsBackendTimsTof <- function() {
   new("MsBackendTimsTof")
-}
\ No newline at end of file
+}
+
+#' Read peaks from a single .d folder.
+#'
+#' @param x `character(1)` indicating the file to read from.
+#'
+#' @noRd
+# .timstof_peaks <- function(x) {
+#   tms <- OpenTIMS(x)
+#   tmp <- query(tms, tms@frames$Id, c("frame", "scan", "mz", "intensity"))
+#   # the result below should be a list of matrices (and not data.frames), right?
+#   unname(split.data.frame(as.matrix(tmp[, c("mz", "intensity")]), 
+#                           factor(paste(tmp$frame, tmp$scan))))
+# }
+
+#' Read columns (mz, intensity or both) from a single file
+#' 
+#' @param x `character(1)` indicating the file to read from.
+#'
+#' @noRd
+.read_frame_col <- function(x, columns) {
+  tms <- OpenTIMS(x)
+  if (any(!columns %in% tms@all_columns))
```

Comment:
Yes, I wanted to add that too but I wasn't sure because manual says that resources are freed anyway but yes it's better to make sure that they are as soon as they are no longer needed. Thanks!